.github/workflows/ci-accounts-service.yml

```diff
@@ -28,13 +30,28 @@ jobs:
         key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
         restore-keys: ${{ runner.os }}-m2
 
-    - name: Build with Maven
-      run: mvn verify --file data-access-services/pom.xml -pl accounts-service -am
-
-    #- name: Publish to GitHub Packages Apache Maven
-    #  run: mvn deploy --file domain-service/credit-application/pom.xml -DskipTests
-    #  env:
-    #    GITHUB_TOKEN: ${{ github.token }}
+    - name: Build and run sonar analysis of the PR
```

Comment:
So we execute the sonar checks for all `data-access-services` in the scope of the `ci-accounts-service.yml` build pipeline right? Though this works in practice, it makes the CI setup a bit confusing since responsibility of the `ci-accounts-service.yml` and `ci-transactions-service.yml` is overlapping (former one doing checks related to the latter).

For a future improvement I'd suggest a multistage build pipeline for data access services with clear separation of jobs of 1) build and verify `data-access-services` 2) sonar scan for `data-access-services` 3-4) In parallel ECR push for `accounts-service` and `transactions-service`. Imho this would make the setup more clear and less redundant.

Reply:
yes, you are right.
technically we don't need the transactions CI anymore apart from the push to ecr.
will create a separate ticket for updating to multi stage.

Reply:
https://deloittedigitalat.visualstudio.com/core-banking/_boards/board/t/core-banking%20Team/Stories/?workitem=2352